src/fam.rs

```diff
@@ -422,9 +422,20 @@ impl<T: Default + FamStruct> FamStructWrapper<T> {
     }
 }
 
-impl<T: Default + FamStruct> PartialEq for FamStructWrapper<T> {
+impl<T: Default + FamStruct + PartialEq> PartialEq for FamStructWrapper<T> {
     fn eq(&self, other: &FamStructWrapper<T>) -> bool {
-        self.as_slice() == other.as_slice()
+        // This is safe as long as the requirements for the `FamStruct` trait to be safe are met
+        // (the implementing type and the entries elements are POD, therefore `Copy`, so memory
+        // safety can't be violated by the ownership of `fam_struct`s). It is also safe because
+        // we're trying to read `T`s from `&T`s that are pointing to properly initialized and
+        // aligned `T`s.
+        let (fam_struct, other_fam_struct): (T, T) = unsafe {
+            (
+                std::ptr::read(self.as_fam_struct_ref()),
```

Comment:
why do we need to std::ptr::read() instead of comparing self.as_fam_struct_ref() and other.as_fam_struct_ref() directly?